src/blocks/content-helper/current-post-details/component.tsx

```diff
@@ -0,0 +1,341 @@
+			case 'other': return __( 'Other', 'wp-parsely' );
+			case 'internal': return __( 'Internal', 'wp-parsely' );
+			case 'direct': return __( 'Direct', 'wp-parsely' );
+		}
+
+		return key;
+	};
+
+	return (
+		<div className="section referrer-types">
+			<div className="section-title">{ __( 'Referrers (Page Views)', 'wp-parsely' ) }</div>
+
+			<div className="multi-percentage-bar">{
+				Object.entries( data.referrers.types ).map( ( [ key, value ] ) => {
+					return (
+						<div className={ 'bar-fill ' + key } key={ key }
+							title={
+								/* translators: 1: Referrer type, 2: Percentage value, %%: Escaped percent sign */
+								sprintf( __( '%s: %s%%', 'wp-parsely' ), getKeyTitle( key ), value.viewsPercentage ) // eslint-disable-line @wordpress/valid-sprintf
+							}
+							style={ { width: value.viewsPercentage + '%' } }>
+						</div>
+					);
+				} ) }
+			</div>
+
+			<table>
+				<thead>
+					<tr>{
+						Object.keys( data.referrers.types ).map( ( key ) => {
+							return <th key={ key }>{ getKeyTitle( key ) }</th>;
+						} ) }
+					</tr>
+				</thead>
+				<tbody>
+					<tr>{
+						Object.entries( data.referrers.types ).map( ( [ key, value ] ) => {
+							return <td key={ key }>{ ImpreciseNumber( value.views ) }</td>;
+						} ) }
+					</tr>
+				</tbody>
+			</table>
+		</div>
+	);
+}
+
+/**
+ * Outputs the "Top Referrers" section.
+ *
+ * @param {PostPerformanceData} data The data needed to populate the section.
+ */
+function TopReferrersSection( data: PostPerformanceData ) {
+	let totalViewsPercentage = 0;
+	return (
+		<div className="section top-referrers">
+			<table>
+				<thead>
+					<tr>
+						<th>{ __( 'Top Referrers', 'wp-parsely' ) }</th>
+						<th colSpan={ 2 }>{ __( 'Page Views', 'wp-parsely' ) }</th>
+					</tr>
+				</thead>
+				<tbody>{
+					Object.entries( data.referrers.top ).map( ( [ key, value ] ) => {
+						if ( key === 'totals' ) {
+							totalViewsPercentage = Number( value.viewsPercentage );
+							return null;
+						}
+
+						let referrerUrl = key;
+						if ( key === 'direct' ) {
+							referrerUrl = __( 'Direct', 'wp-parsely' );
+						}
+
+						return (
+							<tr key={ key }>
+								<th title={ referrerUrl }>{ referrerUrl }</th>
+								<td title={
+									/* translators: %s: Percentage value, %%: Escaped percent sign */
+									sprintf( __( '%s%%', 'wp-parsely' ), value.viewsPercentage ) // eslint-disable-line @wordpress/valid-sprintf
+								}>
+									<div className="percentage-bar" style={ { '--bar-fill': value.viewsPercentage + '%' } as React.CSSProperties }></div>
+								</td>
+								<td>{ ImpreciseNumber( value.views ) }</td>
+							</tr>
+						);
+					} )
+				}
+				</tbody>
+			</table>
+			<div> {
+				/* translators: %s: Percentage value, %%: Escaped percent sign */
+				sprintf( __( '%s%% of views came from top referrers.', 'wp-parsely' ), totalViewsPercentage ) // eslint-disable-line @wordpress/valid-sprintf
```

Comment:
In other locales, the string may change depending on whether the thing being counted is 0, 1, or some other special number, so use https://developer.wordpress.org/block-editor/reference-guides/packages/packages-i18n/#n to allow singular and plural strings, and pass in the `totalViewsPercentage`, and let the i18n system choose the right string for the right number.